terraform/provider/hash.go

```diff
@@ -0,0 +1,58 @@
+package provider
```

Comment:
Is this all novel code you wrote yourself? Or is it based on the hash/signature checks in Terraform? Or OpenTofu? Due to TF's switch to BSL, we have to be careful about what code we use or copy here!

This question applies to `hash.go`, `package_authentication.go`, etc.

Reply:
I took the needed code from Terraform 1.5 and changed it by about 10%. Since we are already importing some terraform code as a package, I'm not sure if I need to additionally specify this in these files.

Reply:
As long as it is MPL TF, that is fine.